### PR TITLE
chore: fix CLI integration tests

### PR DESCRIPTION
**Pull request by rix0rrr**
Fixes:

* Fix breakage introduced by #3782, which changed the wrong string.
* Update other expectation to match change introduced in #3772.

Refactor:

Simplify testing scripts themselves to run commands as they normally would (just runs `cdk`, just calls `npm install @aws-cdk/core`), and remove the dependency on a complex set of sourced bash scripts with functions that need to be called at the right moment.

Instead, there are three new dispatcher scripts:

```
run-against-repo
run-against-dist
run-against-release
```

Which set up an appropriate environment before calling a command that is passed.

Each of them does something different, but through a combination of `PATH` manipulation, config file rewriting and other things makes it so that in the subsequent script the `cdk` command points to an appropriate instance of the CDK, `npm install` and `pip install` will install packages from an appropriate source, etc.

To be used like so:

```
./run-against-repo cli/test.sh
./run-against-dist cli/test.sh
./run-against-release init/test-all.sh
# etc
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/test/integ/cli/test-cdk-ssm-parameter-provider-error.sh [hunk 1]
```diff
@@ -15,7 +15,8 @@ function cdk_synth() {
 }
 
 assert "cdk_synth ${STACK_NAME_PREFIX}-missing-ssm-parameter -c test:ssm-parameter-name=${parameterName}" <<HERE
-SSM parameter not available in account ${account}, region ${region}: ${parameterName}
+[Error at /cdk-toolkit-integration-missing-ssm-parameter] SSM parameter not available in account 993655754359, region us-east-1: /does/not/exist
```
Comment (eladb) [inline packages/aws-cdk/test/integ/cli/test-cdk-ssm-parameter-provider-error.sh]: Are you sure we don’t want to use substitution here?

File: packages/@aws-cdk/cloudformation-diff/lib/format.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ function formatSecurityChangesWithBanner(formatter: Formatter, templateDiff: Tem
   formatter.formatIamChanges(templateDiff.iamChanges);
   formatter.formatSecurityGroupChanges(templateDiff.securityGroupChanges);
 
-  formatter.warning(`(NOTE: There may be security-related changes not in this list. See http://bit.ly/cdk-2EhF7Np)`);
+  formatter.warning(`(NOTE: There may be security-related changes not in this list. See https://github.com/aws/aws-cdk/issues/1299)`);
```
Comment (shivlaks) [inline packages/@aws-cdk/cloudformation-diff/lib/format.ts]: thanks for fixing! I should have caught that

File: packages/aws-cdk/test/integ/cli/test-cdk-ssm-parameter-provider-error.sh (hunk 1 above)
Comment (shivlaks) [reply packages/aws-cdk/test/integ/cli/test-cdk-ssm-parameter-provider-error.sh]: +1

does this work:
```
[Error at ${STACK_NAME_PREFIX}-missing-ssm-parameter] SSM parameter not available in account ${account}, region ${region}: ${parameterName}
```

or adding `errorKey=[Error at /cdk-toolkit-integration-missing-ssm-parameter]` and use it as `${errorKey}`  ahead of the rest of the existing error message?

Comment (rix0rrr) [reply packages/aws-cdk/test/integ/cli/test-cdk-ssm-parameter-provider-error.sh]: Doh, you are correct.

**Comment by rix0rrr** [comment]
I'm going to drastically simplify the scripts used here, and remove the references to verdaccio etc. Setting up the runtime environment so that we can choose what artifacts to work from should be the responsibility of the caller, not of the tests themselves.

**Pull request merged**